presto-docs/src/main/sphinx/develop/functions.rst

```diff
@@ -61,6 +61,17 @@ The function ``is_null`` takes a single ``VARCHAR`` argument and returns a
 the function is of type ``Slice``. ``VARCHAR`` uses ``Slice``, which is essentially
 a wrapper around ``byte[]``, rather than ``String`` for its native container type.
 
+The ``deterministic`` argument indicates that a function has no side effects and,
+for subsequent calls with the same argument(s), the function returns the exact
+same value(s). In Presto, deterministic functions don't rely on any changing state
+and don't modify any state. The ``deterministic`` flag is optional and defaults
+to ``true``. An example for a non-deterministic function is a functions employing
+a random number generator, like :func:`shuffle`. Determinism is only relevant for
```

Comment:
For example, the function :func:`shuffle` is non-deterministic, since it uses random value.